astropy/stats/tests/test_sigma_clipping.py

```diff
@@ -696,3 +696,11 @@ def nan_mad_std(data, axis=None):
     )
 
     assert_allclose(result1, result2)
+
+
+def test_propagation_of_mask():
```

Comment:
Actually, there is a test failure here, because `grow` requires scipy. You need to add `@pytest.mark.skipif(not HAS_SCIPY, reason="test requires scipy")` (assuming the test strictly needs it; if it also does the test without `grow`, I think that would be preferred).

Reply:
Thanks @mhvk! I believe the correct solution is to add the test annotation, since @chris-simpson  noted in https://github.com/astropy/astropy/issues/17401 that the grow parameter is needed to reproduce the problem. I am unable to push to Chris's branch, however, in order to set the tests running (or even send him a PR for some reason), so I'll have to wait for him to add that...

Reply:
BTW, I think this is quite a significant regression, because (eg.) it will change the behaviour of `FittingWithOutlierRemoval` when passed `sigma_clip` as the `outlier_func`, which is typical usage.

Reply:
You can reproduce the issue with `sigma_clipped_stats(y, cenfunc=np.mean, axis=0)` without using the `grow` keyword.  You need a non-default cenfunc or stdfunc and an axis that is not `None`.

Reply:
Re: regression -- Does that mean we should not backport?

Reply:
No, I just meant that the bug addressed by this PR seems fairly significant, although I see that you already have this down for the next patch releases (and presumably it won't make 7.0, which is unfortunate, but we'll just have to constrain DRAGONS as incompatible with 7.0.0 etc.). Thanks!

Reply:
I think a `!=7.0.0` would automatically pull in the next bug fix.

tl;dr -- Does this also mean you don't have this problem in 6.x series?

Reply:
Ah nvm, looking at milestone of offending PR linked, I think you also need a `!=6.1.5`.

Reply:
Yes, that, and also `!=6.1.6`. Having the milestones defined is very helpful in that regard (otherwise we'd have to put `<=6.1.4`).

Reply:
For the 6.x pins, not sure if it matters because now that v7 is out, I think technically we're not obligated to release another 6.x anymore but maybe @astropy/astropy-project-release-team can confirm.

Reply:
(But for consistency, we should still backport to v6.1.x branch, just in case.)

Reply:
7.0 is not yet out so still a bit of time to backport a fix.

Reply:
This test could be rewritten to avoid requiring `scipy` to be installed, but better is the enemy of good here --  I'd rather get this in the 7.0 release.  I'll go ahead and merge.